core-cart.mjs

```diff
@@ -1,10 +1,97 @@
-import {carts, products, promoCode, users} from "./dataset.mjs";
-
-import * as fs from "fs";
-import * as os from "os";
-
-const discountedPrice = (price, rate = 0.10) => (price * (1 - rate)).toFixed(2);
-
-export {
-    discountedPrice,
-};
\ No newline at end of file
+import {carts, products, promoCode, users} from "./dataset.mjs";
+
+import * as fs from "fs";
+import * as os from "os";
+
+const discountedPrice = (price, rate) => (price * (1 - rate)).toFixed(2);
+
+const calculateDiscountedPrice = function (price, rate){
+    return price * (1-rate);
+}
+//cerchiamo lo user per uuid, perciò la freccia qui sotto.
+const getUser = (uuid)=> users.find(user=>user.uuid === uuid);
+//ean lo trovi nel dataset dove ci sono i prodotti
+const getProduct = (productId) => products.find(product => product.ean === productId)
+
+const printShopName = () =>{
+    const machine = os.userInfo();
+    const machineName = machine.username;
+    const pid = process.pid
+    return `   ${machineName.toUpperCase()} - Cart ${pid}`;
+}
+
+const formatData = ()=>{
+    var data = ''
+    var mese = ''
+    
+    var oggi = new Date();

```

Comment:
Basterebbe scrivere una sola riga invece di tutte queste.

La riga magica è: `const today = new Date().toDateString();` documentata qui: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toDateString

Lodevole lo switch ma nella realtà no vedrai nessuno spezzettare cosi la data.


`toLocaleString()` => https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString

potevi usare it-IT => https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locale_identification_and_negotiation